validator/validators/performance/performance_validator.go

```diff
@@ -57,3 +90,153 @@ func (s *PerformanceValidator) Cleanup() error {
 
 	return nil
 }
+
+func (s *PerformanceValidator) SendPacketToDatabase(perfInfo PerformanceInformation) error {
+	var (
+		dataType               = s.vConfig.GetDataType()
+		receiver               = s.vConfig.GetPluginsConfig()
+		commitHash, commitDate = s.vConfig.GetCommitInformation()
+		agentCollectionPeriod  = fmt.Sprint(s.vConfig.GetAgentCollectionPeriod().Seconds())
+		// The secondary global index that is used for checking if there are item has already been exist in the table
+		// The performance validator will query based on the UseCaseHash to confirm if the current commit with the use case
+		// has been exist or not? If yes, merge it. If not, sending it to the database
+		// https://github.com/aws/amazon-cloudwatch-agent-test/blob/e07fe7adb1b1d75244d8984507d3f83a7237c3d3/terraform/setup/main.tf#L46-L53
+		kCheckingAttribute = []string{"CommitHash", "UseCase"}
+		vCheckingAttribute = []string{fmt.Sprint(commitHash), receiver}
+	)
+
+	err := backoff.Retry(func() error {
+		existingPerfInfo, err := awsservice.GetItemInDatabase(DynamoDBDataBase, "UseCaseHash", kCheckingAttribute, vCheckingAttribute, perfInfo)
+		if err != nil {
+			return err
+		}
+
+		// Get the latest performance information from the database and update by merging the existing one
+		// and finally replace the packet in the database
+		maps.Copy(existingPerfInfo["Results"].(map[string]interface{}), perfInfo["Results"].(map[string]interface{}))
+
+		finalPerfInfo := packIntoPerformanceInformation(receiver, dataType, agentCollectionPeriod, commitHash, commitDate, existingPerfInfo["Results"])
+
+		err = awsservice.ReplaceItemInDatabase(DynamoDBDataBase, finalPerfInfo)
+
+		if err != nil {
+			return err
+		}
+		return nil
+	}, awsservice.StandardExponentialBackoff)
+
+	return err
+}
+func (s *PerformanceValidator) CalculateMetricStatsAndPackMetrics(metrics []types.MetricDataResult) (PerformanceInformation, error) {
+	var (
+		receiver               = s.vConfig.GetPluginsConfig()
+		commitHash, commitDate = s.vConfig.GetCommitInformation()
+		dataType               = s.vConfig.GetDataType()
+		dataRate               = fmt.Sprint(s.vConfig.GetDataRate())
+		agentCollectionPeriod  = s.vConfig.GetAgentCollectionPeriod().Seconds()
+	)
+	performanceMetricResults := make(map[string]Stats)
+
+	for _, metric := range metrics {
+		metricLabel := strings.Split(*metric.Label, " ")
```

Comment:
The `metric.Label` will return [human-readable label associated with the data.](https://github.com/aws/aws-sdk-go-v2/blob/main/service/cloudwatch/types/types.go#L747-L748) (e.g `cloudwatch-agent amazon-cloudwatch-agent procstat_cpu_usage`). However, this can be clipped with aggregated dimensions 
```
"aggregation_dimensions": [
			[
				"InstanceId"
			]
]],
```
which will get unclearable when query metrics. However, for future case such as having another file to monitor but using procstat to monitor (e.g performance test will have addition host resources metrics use case which include procstat), this would get unclear and aggregate by CW for the two monitored files by procstat.

Your guys thoughts on this one?

Reply:
Since we control the JSON configuration for the performance tests, we ultimately control aggregation_dimensions and the metric label. So as long as you can clearly document and avoid a future developer turning on aggregation_dimensions and screwing up the performance results... Then I am okay with it.

Reply:
Then I'm cool with it too !!